src/EPR.ProducerContentValidation.Application.UnitTests/Services/Helpers/FindMatchingProducerTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using EPR.ProducerContentValidation.Application.DTOs.SubmissionApi;
+using EPR.ProducerContentValidation.Application.Models;
+using EPR.ProducerContentValidation.Application.Services.Helpers;
+using EPR.ProducerContentValidation.Application.Services.Helpers.Interfaces;
+using EPR.ProducerContentValidation.Data.Models.Subsidiary;
+using EPR.ProducerContentValidation.TestSupport;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace EPR.ProducerContentValidation.Application.UnitTests.Services.Helpers
+{
+    [TestClass]
+    public class FindMatchingProducerTests
+    {
+        private readonly Mock<IOrganisationMatcher> _organisationMatcherMock;
+        private readonly Mock<ISubsidiaryMatcher> _subsidiaryMatcherMock;
+        private readonly Mock<ISubsidiaryValidationEvaluator> _subsidiaryValidationEvaluatorMock;
+        private readonly FindMatchingProducer _findMatchingProducer;
+
+        public FindMatchingProducerTests()
+        {
+            _organisationMatcherMock = new Mock<IOrganisationMatcher>();
+            _subsidiaryMatcherMock = new Mock<ISubsidiaryMatcher>();
+            _subsidiaryValidationEvaluatorMock = new Mock<ISubsidiaryValidationEvaluator>();
+
+            _findMatchingProducer = new FindMatchingProducer(
+                _organisationMatcherMock.Object,
+                _subsidiaryMatcherMock.Object,
+                _subsidiaryValidationEvaluatorMock.Object);
+        }
+
+        [TestMethod]
+        public void Match_NoMatchingOrganisation_ReturnsNull()
+        {
+            // Arrange
+            var row = ModelGenerator.CreateProducerRow(1);
+            var response = new SubsidiaryDetailsResponse();
+
+            _organisationMatcherMock
+                .Setup(m => m.FindMatchingOrganisation(row, response))
+                .Returns((SubsidiaryOrganisationDetail?)null);
+
+            // Act
+            var result = _findMatchingProducer.Match(row, response, 1);
+
+            // Assert
+            Assert.IsNull(result);
```

Comment:
I'd prefer using FluentAssertions .Should().Be... syntax to be consistent with existing tests, as well as some of the other tests you've added.

Just a suggestion - I won't comment on this again in all the tests you have below.